server/core/segment.go

```diff
@@ -0,0 +1,25 @@
+package core
+
+import (
+	"database/sql"
+	"log"
+)
+
+// TODO: normally we shouldn't export members of this domain object.
```

Comment:
Maybe the members could still be exported (if you need them for marshalling) but make `Segment` itself unexported. Depends on what you want to achieve though.